lib/utils/dev-server-flags.js

```diff
@@ -0,0 +1,193 @@
+        },
+        {
+            name: 'key',
+            type: String,
+            describe: 'Path to a SSL key.',
+            group: SSL_GROUP,
+        },
+        {
+            name: 'cert',
+            type: String,
+            describe: 'Path to a SSL certificate.',
+            group: SSL_GROUP,
+        },
+        {
+            name: 'cacert',
+            type: String,
+            describe: 'Path to a SSL CA certificate.',
+            group: SSL_GROUP,
+        },
+        {
+            name: 'pfx',
+            type: String,
+            describe: 'Path to a SSL pfx file.',
+            group: SSL_GROUP,
+        },
+        {
+            name: 'pfx-passphrase',
+            type: String,
+            describe: 'Passphrase for pfx file.',
+            group: SSL_GROUP,
+        },
+        {
+            name: 'content-base',
+            type: String,
+            describe: 'A directory or URL to serve HTML content from.',
+            group: RESPONSE_GROUP,
+        },
+        {
+            name: 'watch-content-base',
+            type: Boolean,
+            describe: 'Enable live-reloading of the content-base.',
+            group: RESPONSE_GROUP,
+        },
+        {
+            name: 'history-api-fallback',
+            type: Boolean,
+            describe: 'Fallback to /index.html for Single Page Applications.',
+            group: RESPONSE_GROUP,
+        },
+        {
+            name: 'compress',
+            type: Boolean,
+            describe: 'Enable gzip compression',
+            group: RESPONSE_GROUP,
+        },
+        // findPort is currently not set up
+        {
+            name: 'port',
+            type: Number,
+            describe: 'The port',
+            group: CONNECTION_GROUP,
+        },
+        {
+            name: 'disable-host-check',
+            type: Boolean,
+            describe: 'Will not check the host',
+            group: CONNECTION_GROUP,
+        },
+        {
+            name: 'socket',
+            type: String,
+            describe: 'Socket to listen',
+            group: CONNECTION_GROUP,
+        },
+        {
+            name: 'public',
+            type: String,
+            describe: 'The public hostname/ip address of the server',
+            group: CONNECTION_GROUP,
+        },
+        {
+            name: 'host',
+            type: String,
+            describe: 'The hostname/ip address the server will bind to',
+            group: CONNECTION_GROUP,
+        },
+        // use command-line-args "multiple" option, allowing the usage: --allowed-hosts host1 host2 host3
+        // instead of the old, comma-separated syntax: --allowed-hosts host1,host2,host3
+        {
+            name: 'allowed-hosts',
+            type: String,
+            describe:
+                'A list of hosts that are allowed to access the dev server, separated by spaces',
+            group: CONNECTION_GROUP,
+            multiple: true,
+        },
+    ],
+};
```

Comment:
/cc @Loonride it is very very very bad, next PR will change a lot of options, we don't need this do in webpack-cli